scanner/src/main/kotlin/Scanner.kt

```diff
@@ -61,20 +61,20 @@ abstract class Scanner(protected val config: ScannerConfiguration) {
     override fun toString(): String = javaClass.simpleName
 
     /**
-     * Scan the list of [packages] using this [Scanner] and store the scan results in [outputDirectory]. If
-     * [downloadDirectory] is specified, it is used instead of [outputDirectory] to download the source code to.
-     * [ScanResult]s are returned associated by the [Package]. The map may contain multiple results for the same
-     * [Package] if the cache contains more than one result for the specification of this scanner.
+     * Scan the list of [packages] and store the scan results in [outputDirectory]. The [downloadDirectory] is used to
+     * download the source code to for scanning. [ScanResult]s are returned associated by the [Package]. The map may
+     * contain multiple results for the same [Package] if the cache contains more than one result for the specification
+     * of this scanner.
      */
-    abstract fun scan(packages: List<Package>, outputDirectory: File, downloadDirectory: File? = null)
+    abstract fun scan(packages: List<Package>, outputDirectory: File, downloadDirectory: File)
             : Map<Package, List<ScanResult>>
 
     /**
-     * Scan the [Project]s and [Package]s specified in [dependenciesFile] using this [Scanner] and store the scan
-     * results in [outputDirectory]. If [downloadDirectory] is specified, it is used instead of [outputDirectory] to
-     * download the source code to. Return scan results as an [OrtResult].
+     * Scan the [Project]s and [Package]s specified in [dependenciesFile] and store the scan results in
+     * [outputDirectory]. The [downloadDirectory] is used to download the source code to for scanning. Return scan
```

Comment:
"The [downloadDirectory] is used to download the source code to for scanning" 

consider instead: Prior to scanning the source code gets downloaded to [downloadDirectory].

Reply:
I'm about to do more scanner refactorings, so let's eventually do that later to avoid conflicts during rebase now.